coderd/templateversions.go

```diff
@@ -44,8 +44,8 @@ func (api *API) templateVersion(rw http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	templateVersion := httpmw.TemplateVersionParam(r)
 
-	job, err := api.Database.GetProvisionerJobByID(ctx, templateVersion.JobID)
-	if err != nil {
+	jobs, err := api.Database.GetProvisionerJobsByIDsWithQueuePosition(ctx, []uuid.UUID{templateVersion.JobID})
+	if err != nil || len(jobs) == 0 {
```

Comment:
I'm not sure I really like needing to shoehorn this new function to fulfil both of the old queries. Do we really need to return the queue position in all of these routes? What if the frontend only fetched them when it knows it's waiting on a build?

Reply:
I think the schema should stay consistent rather than occasionally being there. Unless we observe a performance problem of course.

Reply:
Should we remove the old queries then? Would be nice to have a `:one` and `:many` query as well so you don't have to `|| len(jobs) == 0` all the time. That seems like a pretty easy footgun

Reply:
I did that initially, but figured that was a mistake because we frequently query just to collect perms, where it's a complete waste to get the queue pos.

Reply:
ah makes sense

Reply:
bump
> Would be nice to have a :one and :many query as well so you don't have to || len(jobs) == 0 all the time. That seems like a pretty easy footgun

Reply:
Ahh, I think the opposite, actually. Because the query itself has many joins and such, I think it'd be a mistake to duplicate that.

Reply:
sad that sqlc doesn't have better templating for this